src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -409,14 +409,31 @@ modifyPendingData f = do
       Nothing -> bsPendingBlock %= f
 
 doCreateUserTable :: TableName -> ModuleName -> BlockHandler SQLiteEnv ()
-doCreateUserTable tn@(TableName ttxt) mn =
-    modifyPendingData $ \(c, w, l, p) ->
-        let !c' = HashSet.insert (T.encodeUtf8 ttxt) c
-            !l' = M.insertWith DL.append (TableName txlogKey) txlogs l
-            !t = (c', w, l', p)
-        in t
+doCreateUserTable tn@(TableName ttxt) mn = do
+    -- first check if tablename already exists in pending queues
+    m <- runMaybeT $ checkDbTableExists (Utf8 $ T.encodeUtf8 ttxt)
+    case m of
+      Nothing -> internalError $ dupMsg ttxt
```

Comment:
- We want something more descriptive than `internalError` here maybe?
  - What's up with `Maybe ()`? That's isomorphic to `Bool`. I would _not_ expect a function called `checkDbTableExists` returning `Nothing` to mean "yeah, I found it!"
  - `internalError $ dupMsg ttxt` is duplicated code, refactor into where-block

Reply:
I'm using `checkDbTableExists` in the same manner as you did earlier in the code.